scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -842,9 +831,7 @@ function Install-Packages(
     if ($restartNeeded) {
         Write-HostRed 'Reboot required. To complete the installation after the reboot, run "Initialize-IoTEdge".'
         Restart-Computer -Confirm:(-not $RestartIfNeeded) -Force:$RestartIfNeeded
-    }
-    else {
-        $script:installPackagesCompleted = $true
+        throw
```

Comment:
This throw does not feel right to me. This is not a failure case.

Reply:
The only way to get here would be if the user denied the restart, right? Shouldn't a process exit with non-zero if it asks for permission to do something and then doesn't get it?

Reply:
Maybe. The process is however not failing. It offered to reboot, yet user chose to ignore that request/postpone the reboot. Not a failure per se in my opinion. Up to you.

Reply:
I suppose since `Restart-Computer` itself does not throw when user denies permission, that means PS convention is that the command should succeed in such a case. I'll revert this.